webclient/app/src/app/commons/gitDownloadButton/GitDataButton.jsx

```diff
@@ -0,0 +1,144 @@
+        setAlert({"open":false, "title": "ERROR", "message": ""});
+    }
+
+    const onClose = () => {
+        setDownloadRequestData({ "username": "", "password": "" });
+        setOpenAuthDialog(false);
+    }
+
+    useEffect(() => {
+        if (!downloadRequestData) {
+            return;
+        }
+        let hasError = false;
+
+        if (!RegexConfig.appTemplateAuthUser.test(downloadRequestData.username)) {
+            hasError = true;
+        }
+        if (!RegexConfig.appTemplateAuthPassword.test(downloadRequestData.password)) {
+            hasError = true;
+        }
+        setHasFormError(hasError);
+    }, [downloadRequestData, hasFormError])
+
+
+    return (
+        <Container>
+            <Button onClick={handleLogin} startIcon={buttonIcon ? buttonIcon : <CloudSync />} variant={buttonVariant} >{buttonName}</Button>
+            <NotificationDialog 
+                open={alert.open && !credentialsRequired} 
+                onClose={handleAlertClose} 
+                severity="error" 
+                title={t(alert.title)} 
+                message={t(alert.message)} 
+            />
+            <Dialog open={openAuthDialog} onClose={onClose} spacing={2}>
+                <DialogTitle className={gitDataButtonStyles.dialogHeaderBar}>
+                    <Typography noWrap variant={"h6"} component={"p"}>
+                        {t("appTemplateAuthDialog.title")}
+                    </Typography>
+                    <div className={gitDataButtonStyles.flex} />
+                    <IconButton
+                        aria-label="close"
+                        onClick={onClose}
+                    >
+                        <Close />
+                    </IconButton>
+                </DialogTitle>
+                <Box
+                    component="form"
+                    sx={{
+                        '& .MuiTextField-root': { m: 1, width: '95%' },
+
+                    }}
+                    noValidate
+                    autoComplete="off"
+                >
+                    <ErrorAlert alert={alert} />
+                    <SimpleValidatedTextField
+                        fullWidth
+                        label={t("appTemplateAuthDialog.user") + "*"}
+                        value={downloadRequestData.username}
+                        name="username"
+                        onChange={handleChange}
+                        iscreate={+true}
+                        errortext={t("appTemplateAuthDialog.user.error")}
+                        regex={RegexConfig.appTemplateAuthUser}
+                    />
+                    <SimpleValidatedTextField
+                        type="password"
+                        fullWidth
+                        label={t("appTemplateAuthDialog.password") + "*"}
+                        value={downloadRequestData.password}
+                        name="password"
+                        onChange={handleChange}
+                        iscreate={+true}
```

Comment:
why do you need a + before true?

Reply:
because iscreate: PropTypes.bool is not working properly. See ticket ab#89. Should be done in ticket ab#49 anyway (alining the different validations of app and appTemplate).